src/telemetry.rs

```diff
@@ -7,6 +7,25 @@ use opentelemetry_sdk::{
 
 use crate::SubscriberConfig;
 
+fn normalize_collector_url(collector_url: &str) -> String {
+    // OTLP before version 0.15 didn't append a /v1/traces suffix, but started doing so there.
+    // For backwards compatibility we strip it from configurations that do have it.
+    let collector_url = collector_url
+        // In case of a trailing slash strip it
+        .strip_suffix('/')
+        .unwrap_or(collector_url)
+        .strip_suffix("/v1/traces")
+        .unwrap_or(collector_url);
+
+    // Backport https://github.com/open-telemetry/opentelemetry-rust/pull/1553
+    let collector_url = collector_url.strip_suffix('/').unwrap_or(collector_url);
+
+    // And now starting from version 0.23 opentelemetry randomly stopped appending
+    // the url suffix, so we need to do it ourselves.
+    // This was not announced in the changelogs 🙃
+    collector_url.to_string() + "/v1/traces"
```

Comment:
Are we stripping it and add it back again...? :thinking: Am I being dumb?

Reply:
Yeah we need to ensure that it's always there and the easiest way to do that is to remove it first if the user already added it manually and then remove it